src/include/main/database.h

```diff
@@ -98,6 +98,7 @@ class Database {
 private:
     std::string databasePath;
     SystemConfig systemConfig;
+    std::unique_ptr<common::FileSystem> fs;
```

Comment:
Should we store this field as `VirtualFileSystem`? It is more readable, as this is the only possible instantiation, at least for quite a while.